src/components/Modals/WebcamModal.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useRef } from 'react';
+import Webcam from 'react-webcam';
+import Modal from '@mui/material/Modal';
+import Button from '@mui/material/Button';
+
+// web camera modal
+const WebcamModal = ({ open, onClose, onCapture }) => {
+  const webcamRef = useRef(null);
+
+  const handleCapture = () => {
+    const imageSrc = webcamRef.current.getScreenshot();
+    onCapture(imageSrc);
+    onClose();
+  };
+
+  return (
+    <Modal
+      open={open}
+      onClose={onClose}
+      aria-labelledby="webcam-modal-title"
+      aria-describedby="webcam-modal-description"
+    >
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          height: '100vh'
+        }}
+      >
+        <h2 id="webcam-modal-title">Webcam</h2>
+        <Webcam
+          audio={false}
+          ref={webcamRef}
+          screenshotFormat="image/jpeg"
+          videoConstraints={{ facingMode: 'user' }}
+        />
+        <Button variant="contained" color="primary" onClick={handleCapture}>
```

Comment:
Although this is a good start, was wondering if there could be another button that could be created for simply exiting "webcam mode"?

Reply:
Looks like you can escape using the `esc` key on the keyboard. I'm resolving this.